openapi.yaml

```diff
@@ -0,0 +1,220 @@
+      required: true
+      schema:
+        # CVE ID format
+        type: string
+        pattern: 'CVE-\d{4}-\d+'
+    api-entity-header:
+      name: X-API-ENTITY
+      description: >
+        The entity (e.g., CNA, ADP) that is authenticated or requesting authentication. \[TODO:: Is there a pattern for the format?\]
+      in: header
+      required: true
+      schema:
+        type: string
+    api-secret-header:
+      name: X-API-SECRET
+      description: >
+        The entity's (e.g., CNA, ADP) secret. \[TODO:: Is there a pattern for the format?\]
+      in: header
+      required: true
+      schema:
+        type: string
+paths:
+  # CVE submission and retrieval service
+  /cve/{cve-id}:
+    parameters:
+      - "$ref": "#/components/parameters/cve-id-path"
+    get:
+      summary: Retrieves a CVE record by id
+      operationId: getCVERecordById
+      responses:
+        200:
+          description: >
+            The requested CVE record associated with the CVE ID is returned.
+          content:
+            application/json:
+              schema:
+                type: object
+                # add schema for CVE 5.0 format
+        404:
+          description: The requested CVE ID has not been populated.
+    post:
+      summary: Creates a new CVE record entry
+      operationId: submitCVERecordById
+      parameters:
+        - "$ref": "#/components/parameters/api-entity-header"
+        - "$ref": "#/components/parameters/api-secret-header"
+        - name: X-API-AUTHOR
+          in: header
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: The CVE record data to populate
+        content:
+          application/json:
+            schema:
+              type: object
+              # add schema for CVE 5.0 format
+      responses:
+        200:
+          description: the CVE record is posted
+          content:
+            application/json:
+              schema:
+                type: object
+                # schema for the body
+        400:
+          description: the request body is malformed
+        401:
+          description: the client is not authenticated
+        403:
+          description: >
+            the client is not authorized to submit this CVE. A likely
+            cause is the client's authentication context is not
+            associated with the reserved CVE id.
+        404:
+          description: >
+            the requested CVE ID has not been reserved \[TODO:: do we want
+            this as it leaks info about the reserved status of a CVE ID? \]
+    put:
+      summary: Updates an existing CVE record entry
+      operationId: updateCVERecordById
+      parameters:
+        - "$ref": "#/components/parameters/api-entity-header"
+        - "$ref": "#/components/parameters/api-secret-header"
+        - name: X-API-AUTHOR
+          in: header
+          required: true
+          schema:
+            type: string
```

Comment:
Pulled this out of the source code. Is this needed? Or can this be determined by the authentication context?